packages/compiler-core/src/transforms/vModel.ts

```diff
@@ -130,26 +134,80 @@ export const transformModel: DirectiveTransform = (dir, node, context) => {
 
   // modelModifiers: { foo: true, "bar-baz": true }
   if (dir.modifiers.length && node.tagType === ElementTypes.COMPONENT) {
-    const modifiers = dir.modifiers
-      .map(m => m.content)
-      .map(m => (isSimpleIdentifier(m) ? m : JSON.stringify(m)) + `: true`)
-      .join(`, `)
+    const trueExpression = createSimpleExpression(
+      `true`,
+      false,
+      dir.loc,
+      ConstantTypes.CAN_CACHE,
+    )
+
+    const staticMods: ExpressionNode[] = []
+    const callArgs: (ObjectExpression | ExpressionNode)[] = []
+
     const modifiersKey = arg
       ? isStaticExp(arg)
         ? `${arg.content}Modifiers`
         : createCompoundExpression([arg, ' + "Modifiers"'])
       : `modelModifiers`
-    props.push(
-      createObjectProperty(
-        modifiersKey,
-        createSimpleExpression(
-          `{ ${modifiers} }`,
-          false,
-          dir.loc,
-          ConstantTypes.CAN_CACHE,
-        ),
-      ),
-    )
+
+    for (let i = 0; i < dir.modifiers.length; i++) {
```

Comment:
The code here seems to be the same as in transformsElement.ts and should be considered for reuse